packages/enzyme-adapter-react-16.3/src/ReactSixteenThreeAdapter.js

```diff
@@ -312,6 +316,12 @@ class ReactSixteenThreeAdapter extends EnzymeAdapter {
         return fn();
         // return ReactDOM.unstable_batchedUpdates(fn);
       },
+      setProviders(newProviders) {
+        if (newProviders.some(provider => !isContextProvider(provider))) {
+          throw new Error('ReactWrapper::setProviders() may only be called with Provider elements');
+        }
+        instance.setProviders(newProviders);
+      },
```

Comment:
It seems like the more idiomatic way to do this would be to handle the update via the `render()` method. However, I'm pretty sure (correct me if I'm wrong) changing the signature of that method would need to force this feature into a new major version.

Adding a `setProviders()` method here at least allows enzyme to throw a helpful error if react/the adapter is at too old of a version.